actix-server/src/worker.rs

```diff
@@ -38,35 +38,114 @@ pub(crate) struct Stop {
 #[derive(Debug)]
 pub(crate) struct Conn {
     pub io: MioStream,
-    pub token: Token,
+    pub token: usize,
 }
 
 fn handle_pair(
     idx: usize,
     tx1: UnboundedSender<Conn>,
     tx2: UnboundedSender<Stop>,
-    avail: WorkerAvailability,
+    counter: Counter,
 ) -> (WorkerHandleAccept, WorkerHandleServer) {
-    let accept = WorkerHandleAccept { tx: tx1, avail };
+    let accept = WorkerHandleAccept {
+        idx,
+        tx: tx1,
+        counter,
+    };
 
     let server = WorkerHandleServer { idx, tx: tx2 };
 
     (accept, server)
 }
 
+#[derive(Clone)]
+pub(crate) struct Counter {
+    counter: Arc<AtomicUsize>,
+    limit: usize,
+}
+
+impl Counter {
+    pub(crate) fn new(limit: usize) -> Self {
+        Self {
+            counter: Arc::new(AtomicUsize::new(1)),
```

Comment:
fetch_xxx atomic operation would wrap around on overflow. So unless we add a stronger order and do an assert there is no way to check overflow.

Reply:
i didn't know about the wrapping behaviour

Reply:
I can add an debug assert but I'm not sure if Relaxed order would work as we could have a lot of tasks on worker thread trying to decrement the counter.

Reply:
I think the syncronisation comes from elsewhere but I guess I'd feel more comfortable if they were AcqRel.

Reply:
I would say we just use this and the worst case is over limiting cause a worker to hang.(I doubt it can happen but who knows). 

We can improve it when there is report on worker hang with very high(or little) workload coupled with very low worker limit.